saw-core/src/Verifier/SAW/Simulator/Prims.hs

```diff
@@ -808,6 +816,8 @@ expByNatOp bp =
 
          loop one . V.toList =<< toBits (bpUnpack bp) w
 
+    VIntToNat _ -> error "expByNat: symbolic integer"
```

Comment:
I guess exponentiation by a symbolic integer will never be supported, as it's not in the SMT-Lib theory. We should throw a catchable user-error exception.